server/queries/user/checkEmailExistsQuery.ts

```diff
@@ -0,0 +1,12 @@
+import connection from '../../database/connection';
+
+const checkEmailExistsQuery = (
+  email: String,
+) => {
+  const sql = {
+    text: 'select * from users where email = $1',
```

Comment:
If you want this only for checking if the email exists or not then you don't need to return all columns with `'*'`, Selecting the id only would be enough.
```suggestion
    text: 'select id from users where email = $1',
```